types/external/gfycat.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
Added types for `gfycat-sdk`. Source of the type is [this repo](https://github.com/sylv/gfycat-api/blob/885cbaf59e08e9d36bfb1b687a72d120d18b08c4/src/types.ts#L11). 

I just added `GfycatAPITag` and  cross-checked `GfycatAPIItem` and made some items undefined with `?` based on the response I received from API. 

One question here: I see that there is already type file [gfycat-sdk.d.ts](https://github.com/mattermost/mattermost-webapp/blob/master/types/external/gfycat-sdk.d.ts). How can I combine these two files? 

When I just copy types from this file to `gfycat-sdk.d.ts` file, I receive  `Could not find a declaration file for module 'gfycat-sdk'. '/workspace/mattermost-webapp/node_modules/gfycat-sdk/lib/index.js' implicitly has an 'any' type.` error

Reply:
I think there's some weird requirements for `.d.ts` files to have them get automatically detected by TypeScript. I ran into something similar with another PR where importing types in a `.d.ts` file caused it to not be detected [unless I imported them inside the module I was declaring](https://github.com/mattermost/mattermost-webapp/blob/MM-46004_module-federation/types/products.d.ts).

Perhaps you could declare these inside the `gfycat-sdk` module to make that work? Like
```ts
declare module 'gfycat-sdk' {
    export type GfycatAPIItem {...}
}
```

If not, keeping them in a separate file works just about as well

Reply:
I also tried that but this I receive `no initializer` related error. Anyhow, as mentioned in the previous comment I added external types under types package